Compare with v6.6.0, v7.0.0 has 140%-180% performance regression with errors on sysbench oltp_write_only workload

### What did you do?

1. deploy 2 TiDB and cdc
2. run sysbench oltp_write_only

### What did you expect to see?

No performance regression, the cdc_changefeed_checkpoint_lag is similar.

### What did you see instead?

![image](https://user-images.githubusercontent.com/84501897/226775372-885f1602-fe5f-4b1f-9fcb-155a83862231.png)


### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
v7.0.0
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
v7.0.0
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
v7.0.0
```

Current investigation:
Found a lot of errors like the following in cdc.log:
```
[2023/03/22 *13:15:10.526 +08:00*] [ERROR] [packets.go:37] ["read tcp 10.233.80.158:43552->10.106.26.249:4000: i/o timeout"] [component="[mysql]"]
```
This indicates that the connection between cdc and downstream TiDB is often disconnected, causing sink to fail to write data downstream. Currently, it is suspected that the load balance service in the test environment is the problem.

- From benchmark, we can observer ticdc met following error
```go
[2023/03/22 19:25:45.304 +08:00] [ERROR] [packets.go:37] ["read tcp 10.233.80.156:42700->10.109.103.19:4000: i/o timeout"] [component="[mysql]"]
[2023/03/22 19:25:45.305 +08:00] [WARN] [mysql.go:819] ["execute DMLs with error, retry later"] [error="[CDC:ErrMySQLTxnError]MySQL txn error: invalid connection"] [errorVerbose="[CDC:ErrMySQLTxnError]MySQL txn error: invalid connection\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/errors.WrapError\n\tgithub.com/pingcap/tiflow/pkg/errors/helper.go:34\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).multiStmtExecute\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:661\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries.func1.3\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:761\ngithub.com/pingcap/tiflow/cdc/sink/metrics.(*Statistics).RecordBatchExecution\n\tgithub.com/pingcap/tiflow/cdc/sink/metrics/statistics.go:74\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries.func1\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:747\ngithub.com/pingcap/tiflow/pkg/retry.run\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:57\ngithub.com/pingcap/tiflow/pkg/retry.Do\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:34\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:734\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).Flush\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:228\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.(*worker).doFlush\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/worker.go:179\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.(*worker).runLoop\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/worker.go:135\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.newSink.func1\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/txn_dml_sink.go:103\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_arm64.s:1172"] [duration=2m0.039613308s] [query=""] [count=78] [startTs="["] [changefeed=default.simple-replication-task]
```
- But peer TiDB doesn't have following connection abort log

```go
[2023/02/21 02:08:07.947 +08:00] [INFO] [conn.go:1143] ["command dispatched failed"] [conn=1408063177832271097] [connInfo="id:1408063177832271097, addr:[10.2.7.36:52218](http://10.2.7.36:52218/) status:10, collation:utf8mb4_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql=COMMIT] [txn_mode=OPTIMISTIC] [timestamp=439593514555670529] [err="write tcp [10.2.6.127:14000](http://10.2.6.127:14000/)->[10.2.7.36:52218](http://10.2.7.36:52218/): write: broken pipe
```

- It seems the data doesn't send to peer TiDB, maybe kept in send buffer of TiCDC node because of some network problem.

--- a/pkg/sink/mysql/config.go
+++ b/pkg/sink/mysql/config.go
@@ -72,7 +72,7 @@ const (
 	BackoffMaxDelay = 60 * time.Second
 
 	defaultBatchDMLEnable  = true
-	defaultMultiStmtEnable = true
+	defaultMultiStmtEnable = false
 
 	// defaultcachePrepStmts is the default value of cachePrepStmts
 	defaultCachePrepStmts = true

